homeassistant/components/knx/sensor.py

```diff
@@ -1,77 +1,42 @@
 """Support for KNX/IP sensors."""
 from xknx.devices import Sensor as XknxSensor
 
-from homeassistant.core import callback
+from homeassistant.components.sensor import DEVICE_CLASSES
 from homeassistant.helpers.entity import Entity
 
-from . import DATA_KNX
+from .const import DOMAIN
+from .knx_entity import KnxEntity
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up sensor(s) for KNX platform."""
     entities = []
-    for device in hass.data[DATA_KNX].xknx.devices:
+    for device in hass.data[DOMAIN].xknx.devices:
         if isinstance(device, XknxSensor):
             entities.append(KNXSensor(device))
     async_add_entities(entities)
 
 
-class KNXSensor(Entity):
+class KNXSensor(KnxEntity, Entity):
     """Representation of a KNX sensor."""
 
     def __init__(self, device: XknxSensor):
         """Initialize of a KNX sensor."""
-        self.device = device
-
-    @callback
-    def async_register_callbacks(self):
-        """Register callbacks to update hass after device was changed."""
-
-        async def after_update_callback(device):
-            """Call after device was updated."""
-            self.async_write_ha_state()
-
-        self.device.register_device_updated_cb(after_update_callback)
-
-    async def async_added_to_hass(self):
-        """Store register state change callback."""
-        self.async_register_callbacks()
-
-    async def async_update(self):
-        """Update the state from KNX."""
-        await self.device.sync()
-
-    @property
-    def name(self):
-        """Return the name of the KNX device."""
-        return self.device.name
-
-    @property
-    def available(self):
-        """Return True if entity is available."""
-        return self.hass.data[DATA_KNX].connected
-
-    @property
-    def should_poll(self):
-        """No polling needed within KNX."""
-        return False
+        super().__init__(device)
 
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self.device.resolve_state()
+        return self._device.resolve_state()
 
     @property
     def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
-        return self.device.unit_of_measurement()
+        return self._device.unit_of_measurement()
 
     @property
     def device_class(self):
         """Return the device class of the sensor."""
-        return self.device.ha_device_class()
-
-    @property
-    def device_state_attributes(self):
-        """Return the state attributes."""
+        if self._device.ha_device_class() in DEVICE_CLASSES:
```

Comment:
Maybe cache the device class in a local variable first? We'll save one function call.

Reply:
I can change that, but honestly - `ha_device_class` should be a property. We'll change that in the next update of the library. 

```
    def ha_device_class(self):
        """Return the home assistant device class as string."""
        return self.sensor_value.ha_device_class
```

Do you still want me to cache it?

Reply:
Done